blob/gcsblob/gcsblob.go

```diff
@@ -120,6 +121,7 @@ func (b *bucket) NewTypedWriter(ctx context.Context, key string, contentType str
 	w.ContentType = contentType
 	if opts != nil {
 		w.ChunkSize = bufferSize(opts.BufferSize)
+		w.Metadata = opts.Metadata
```

Comment:
This should throw an error if the metadata contains a key like `Content-Type`.

Reply:
Why? It seems to work fine.... The providers namespace the keys, so they don't conflict with real HTTP headers/ attributes.

For example, S3 does this:

```
Content-Type:
      - text/plain; charset=utf-8
...
X-Amz-Meta-Content-Type:
      - value-a
```

and GCS passes the metadata in the body, not in an HTTP header (at least for the request we're using; for other requests they are headers, but have a prefix similar to what S3 does).